src/os/ui/layer/compare/layercompareui.js

```diff
@@ -549,6 +609,77 @@ export class Controller {
     }
   }
 
+  /**
+   * Removes a layer from the compare.
+   * @param {Layer|ILayer} layer The layer to remove.
+   */
+  remove(layer) {
+    // please the compiler and our terrible layer typing
+    layer = /** @type {Layer} */ (layer);
+
+    const rightLayerArr = this.rightLayers.getArray().slice();
```

Comment:
Nit, but as long as we're always creating a new array we could combine a few lines here with:
```
this.rightLayers.getArray().filter((item) => item !== layer);
```